test/src/test/java/org/corfudb/integration/SecurityIT.java

```diff
@@ -0,0 +1,220 @@
+    private String serverPathToTrustStore;
+    private String serverPathToTrustStorePassword;
+    private String runtimePathToKeyStore;
+    private String runtimePathToKeyStorePassword;
+    private String runtimePathToTrustStore;
+    private String runtimePathToTrustStorePassword;
+
+    /* A helper method that start a single TLS enabled server and returns a process. */
+    private Process runSinglePersistentServerTls() throws IOException {
+        return new AbstractIT.CorfuServerRunner()
+                .setHost(corfuSingleNodeHost)
+                .setPort(corfuStringNodePort)
+                .setTlsEnabled(tlsEnabled)
+                .setKeyStore(serverPathToKeyStore)
+                .setKeyStorePassword(serverPathToKeyStorePassword)
+                .setTrustStore(serverPathToTrustStore)
+                .setTrustStorePassword(serverPathToTrustStorePassword)
+                .setLogPath(getCorfuServerLogPath(corfuSingleNodeHost, corfuStringNodePort))
+                .setSingle(true)
+                .runServer();
+    }
+
+    /**
+     * This is a helper method that load the properties required for creating corfu servers before
+     * each test by converting them from the values provided in CorfuDB.properties. Note that it
+     * will throw {@link NumberFormatException} if the numerical properties provided in CorfuDB.properties
+     * are not well-formed.
+     */
+    @Before
+    public void loadProperties() {
+        // Load host and port properties
+        corfuSingleNodeHost = PROPERTIES.getProperty("corfuSingleNodeHost");
+        corfuStringNodePort = Integer.parseInt(PROPERTIES.getProperty("corfuSingleNodePort"));
+
+        singleNodeEndpoint = String.format("%s:%d",
+                corfuSingleNodeHost,
+                corfuStringNodePort);
+
+        // Load TLS configuration, keystore and truststore properties
+        tlsEnabled = Boolean.valueOf(PROPERTIES.getProperty("tlsEnabledSecurityIT"));
+        serverPathToKeyStore = getPropertyAbsolutePath("serverPathToKeyStore");
+        serverPathToKeyStorePassword = getPropertyAbsolutePath("serverPathToKeyStorePassword");
+        serverPathToTrustStore = getPropertyAbsolutePath("serverPathToTrustStore");
+        serverPathToTrustStorePassword = getPropertyAbsolutePath("serverPathToTrustStorePassword");
+        runtimePathToKeyStore = getPropertyAbsolutePath("runtimePathToKeyStore");
+        runtimePathToKeyStorePassword = getPropertyAbsolutePath("runtimePathToKeyStorePassword");
+        runtimePathToTrustStore = getPropertyAbsolutePath("runtimePathToTrustStore");
+        runtimePathToTrustStorePassword = getPropertyAbsolutePath("runtimePathToTrustStorePassword");
+    }
+
+    // Take the property string provided int CorfuDB.properties and return the absolute path string
+    private String getPropertyAbsolutePath(String pathProperty) {
+        return Paths.get(PROPERTIES.getProperty(pathProperty)).toAbsolutePath().toString();
```

Comment:
maybe you need to deal with missing or not defined properties exceptions

Reply:
Done. Added logic for dealing with non existent or invalid properties.